Bottom: 51c7753c940fd3727b8cc3e93553c57f89d1d9d2
Top:    e3b6b53e8c087ce585281511e53d57ff0c384975
Author: Jason Wang <jasowang@redhat.com>
Date:   2018-01-22 17:07:03 +0800

virtio-net: add missing virtqueue kick when flushing packets

We tends to batch submitting packets during XDP_TX. This requires to
kick virtqueue after a batch, we tried to do it through
xdp_do_flush_map() which only makes sense for the case of redirection
not XDP_TX. So explicit kick the virtqueue in this case.

Signed-off-by: Jason Wang <jasowang@redhat.com>


---

diff --git a/drivers/net/virtio_net.c b/drivers/net/virtio_net.c
index 626c273..e027e49 100644
--- a/drivers/net/virtio_net.c
+++ b/drivers/net/virtio_net.c
@@ -1243,8 +1243,10 @@ static int virtnet_poll(struct napi_struct *napi, int budget)
 	if (received < budget)
 		virtqueue_napi_complete(napi, rq->vq, received);
 
-	if (xdp_xmit)
+	if (xdp_xmit) {
+		virtqueue_kick(rq->vq);
 		xdp_do_flush_map();
+	}
 
 	return received;
 }
